featuregate/registry.go

```diff
@@ -110,8 +110,8 @@ func (r *Registry) Register(id string, stage Stage, opts ...RegisterOption) (*Ga
 	default:
 		return nil, fmt.Errorf("unknown stage value %q for gate %q", stage, id)
 	}
-	if g.stage == StageStable && g.toVersion == "" {
-		return nil, fmt.Errorf("no removal version set for stable gate %q", id)
+	if (g.stage == StageStable || g.stage == StageDeprecated) && g.toVersion == "" {
```

Comment:
Sorry for creating a bit more work to you, but would really appreciate a conclusion on https://github.com/open-telemetry/opentelemetry-collector/issues/7621, do you mind getting that to the final line?

Reply:
Yup I was planning to work on that issue

Reply:
Lets merge https://github.com/open-telemetry/opentelemetry-collector/pull/7626 before this PR.

Reply:
Please rebase